sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2690,6 +2690,19 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+
+  val GROUPING_WITH_UNION = buildConf("spark.sql.optimizer.grouping")
+    .doc("When true,enable grouping with union.")
+    .booleanConf
+    .createWithDefault(false)
+
+  val GROUPING_EXPAND_PROJECTIONS = buildConf("spark.sql.optimizer.projections.size")
+    .doc("Split a expand into several smallExpand,which contains the size of projections.It can" +
+      "reduce the shuffle pressure and improve performance in multidimensional analysis " +
+      "when data is huge.")
+    .intConf
+    .createWithDefault(4)
```

Comment:
Could you check if all the existing tests pass if `spark.sql.optimizer.grouping`=true and `spark.sql.optimizer.projections.size`=1?

Reply:
ok!